### ISSUE TITLE
Conflicting updates error while inserting into sm-Customer-notification collection as part of merging

### ISSUE THREAD

**Issue by venuiyengar** [marklogic/marklogic-data-hub#3413]
I was using quick start and the tutorial project to see the contents of notification collection
a) I updated the flow to add notify if the threshold is above 15:

          "thresholds" : {
            "threshold" : [ {
              "above" : "25",
              "label" : "HighConfidence",
              "action" : "merge"
            }, {
              "above" : "15",
              "label" : "Notify",
              "action" : "notify"
            } ]
          },
b) I updated an existing new auto insurance customer and removed email for one of the customer so that it generated notification
c) First time through after ingesting, mapping, findDuplicates and merging it added that to the 'sm-Customer-notification' collection as expected
d) I repeated the same steps with the same customer as a new file and it generated this error
e) I tried another customer by removing the email and it still generated this same error

Apparently 'sm-Customer-notification' does not accumulate notifications when they are generated due to this error.

com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-CONFLICTINGUPDATES: xdmp.documentInsert("/com.marklogic.smart-mastering/matcher/notifications/64a21f6e3c6...", cts.doc("/com.marklogic.smart-mastering/matcher/notifications/64a21f6e3c6ee80ab7c1b7f9cc5fb868.xml"), {permissions:[], collections:["mergeDuplicates", "Customer", "sm-Customer-notification", ...], metadata:{datahubCreatedBy:"admin", datahubCreatedByStep:"default-merging", datahubCreatedInFlow:"Sunrise", ...}}) -- Conflicting updates xdmp.documentInsert("/com.marklogic.smart-mastering/matcher/notifications/64a21f6e3c6ee80ab7c1b7f9cc5fb868.xml", cts.doc("/com.marklogic.smart-mastering/matcher/notifications/64a21f6e3c6ee80ab7c1b7f9cc5fb868.xml"), (), ("mergeDuplicates", "Customer", "sm-Customer-notification", "sm-Customer-mastered"), 0) and xdmp.documentInsert("/com.marklogic.smart-mastering/matcher/notifications/64a21f6e3c6ee80ab7c1b7f9cc5fb868.xml", Element("<sm:notification xmlns:sm='http://marklogic.com/smart-mastering'>...</sm:notification>"), (), ("mergeDuplicates", "Customer", "sm-Customer-notification"), 0)

**Comment by xnikhil08** [comment]
Hi @venuiyengar, could you please provide the steps (along with any sample document and data hub flow and step artifacts) to reproduce this issue at our end ?

**Comment by rjrudin** [comment]
@venuiyengar When you share the project, you don't need to share the data directory since that's part of the tutorial project. But we'll need more information on the exact changes you're making in step "b" - i.e. exactly which customer documents are you updating?

**Comment by venuiyengar** [comment]
@rjrudin I took auto customer cust100 and modified to remove email and ran through the flow to generate notify event. Then I created a new customer file using the same file and info and that generated the conflicting update error.

Then I tried modifying and removing email for another customer and it generated the same error.

**Comment by rjrudin** [comment]
Can you zip up your project, minus the data files, and then attach it to this ticket? I want to be sure we're using the exact same config (note that we have a Gradle task planned in the near future that will create this zip for you, but you have to make it manually for now).

**Comment by venuiyengar** [comment]
Here is the zip file, I removed tmp, build, data, .gradle before zipping
I had to remove the war file as well.
[quickstart-tutorial.zip](https://github.com/marklogic/marklogic-data-hub/files/4019961/quickstart-tutorial.zip)

**Comment by rjrudin** [comment]
Thanks, I'm checking this out now.

**Comment by rjrudin** [comment]
I need to clarify these statements:

"I took auto customer cust100 and modified to remove email and ran through the flow to generate notify event. Then I created a new customer file using the same file and info and that generated the conflicting update error."

Does that mean you made e.g. cust101.json with the same info as cust100.json, minus an email address? And then you ran the entire flow, such that a notify document was created? And then you made an e.g. cust102.json file with the same info as cust100.json, and then ran the entire flow again? 

I was able to get a conflicting updates error by doing that, but I want to first make sure that I'm doing the same steps that you are.

**Comment by venuiyengar** [comment]
Yes, that is what I did.

**Comment by rjrudin** [comment]
@venuiyengar  I believe the issue here is that the matching step is both querying on the "Customer" collection, and it's adding match summary documents to the "Customer" collection, resulting in match summary documents being included in the query for the matching step. I removed "Customer" from the "collections" property for the matching step (step 5), and I did not get the conflicting updates error. 

@ryanjdew Does the explanation above sound right to you? Seems similar to DHFPROD-3624?

@venuiyengar Did you add "Customer" to the "collections" array on the matching step 5, or did QuickStart do that automatically?
